### PR TITLE
For pushing ibm-vpc-csi-driver images

### PR DESCRIPTION
**Pull request by ambiknai**

### CODE REVIEW

**Comment by ambiknai** [comment]
@ameukam @cblecker @spiffxp Could some one please review and merge this PR

File: infra/gcp/infra.yaml [hunk 1]
```diff
@@ -325,3 +325,4 @@ infra:
       k8s-staging-storage-migrator:
       k8s-staging-test-infra:
       k8s-staging-txtdirect:
+      k8s-staging-cloud-provider-ibm:
```
Comment (ameukam) [inline infra/gcp/infra.yaml:328]: @ambiknai Please, respect the alphabetic order of this list.

Comment (ambiknai) [reply infra/gcp/infra.yaml]: I have moved it as per this comment.

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Comment by ambiknai** [comment]
@spiffxp  Thanks for approving the PR
I could not see the project when I checked Google Cloud Dashboard.

**Comment by ameukam** [comment]
@ambiknai https://console.cloud.google.com/home/dashboard?project=k8s-staging-capi-ibmcloud

**Comment by ambiknai** [comment]
@ameukam shouldnt it be `k8s-staging-cloud-provider-ibm`

Not sure if I missed something here. I just changed the above url to point to `k8s-staging-cloud-provider-ibm` project and it says permission denied

Screenshot 2021-09-09 at 9 45 31 AM

**Comment by ameukam** [comment]
> @ameukam shouldnt it be `k8s-staging-cloud-provider-ibm`
> 
> Not sure if I missed something here. I just changed the above url to point to `k8s-staging-cloud-provider-ibm` project and it says permission denied
> 
> Screenshot 2021-09-09 at 9 45 31 AM

@ambiknai Sorry wrong project ID: https://console.cloud.google.com/home/dashboard?project=k8s-staging-cloud-provider-ibm.

**Comment by ambiknai** [comment]
> > @ameukam shouldnt it be `k8s-staging-cloud-provider-ibm`
> > Not sure if I missed something here. I just changed the above url to point to `k8s-staging-cloud-provider-ibm` project and it says permission denied
> > Screenshot 2021-09-09 at 9 45 31 AM
> 
> @ambiknai Sorry wrong project ID: https://console.cloud.google.com/home/dashboard?project=k8s-staging-cloud-provider-ibm.

But I dont have permission as shared in screenshot.

**Comment by ameukam** [comment]
Forgot to mention earlier I ran `./ensure-staging-storage.sh k8s-staging-cloud-provider-ibm` instead of
`./ensure-staging-storage.sh cloud-provider-ibm`. 

It's possible the groups reconciliation didn't happen for `k8s-infra-staging-cloud-provider-ibm@kubernetes.io`.

**Comment by ambiknai** [comment]
@ameukam Ok. So how can this be fixed. I still don't see project in my dashboard

**Comment by spiffxp** [comment]
Somehow when I ran `./infra/gcp/bash/ensure-staging-storage.sh cloud-provider-ibm` when this PR merged it failed out at GCR repo creation.

I'm re-running for all staging projects now just to make sure everything is up to date, and I caught this.  You should be good to go now

```
Configuring staging project: k8s-staging-cloud-provider-ibm
  Ensuring project exists: k8s-staging-cloud-provider-ibm
  Ensuring k8s-infra-staging-cloud-provider-ibm@kubernetes.io are project viewers
  Ensuring necessary enabled services staging project: k8s-staging-cloud-provider-ibm
  Ensuring disabled services for staging project: k8s-staging-cloud-provider-ibm
  Ensuring serviceAccount:k8s-infra-gcr-vuln-scanning@k8s-artifacts-prod.iam.gserviceaccount.com can view vulnernability scanning results for project: k8s-staging-cloud-provider-ibm
  Ensuring staging GCR repo: gcr.io/k8s-staging-cloud-provider-ibm
    Ensuring a GCR repo exists for project: k8s-staging-cloud-provider-ibm
    Using default tag: latest
    latest: Pulling from pause
    Digest: sha256:a78c2d6208eff9b672de43f880093100050983047b7b0afe0217d3656e1b0d5f
    Status: Image is up to date for k8s.gcr.io/pause:latest
    k8s.gcr.io/pause:latest
    Using default tag: latest
    The push refers to repository [gcr.io/k8s-staging-cloud-provider-ibm/ceci-nest-pas-une-image]
    5f70bf18a086: Preparing
    e16a89738269: Preparing
    5f70bf18a086: Preparing
    5f70bf18a086: Layer already exists
    e16a89738269: Layer already exists
    latest: digest: sha256:ec3ca3ee90e4dafde96c83232b30f17b5e8992ff35479d0661b8f4ff2f21bf74 size: 938
    Digests:
    - gcr.io/k8s-staging-cloud-provider-ibm/ceci-nest-pas-une-image@sha256:ec3ca3ee90e4dafde96c83232b30f17b5e8992ff35479d0661b8f4ff2f21bf74
      Associated tags:
     - latest
    Tags:
    - gcr.io/k8s-staging-cloud-provider-ibm/ceci-nest-pas-une-image:latest
    Deleted [gcr.io/k8s-staging-cloud-provider-ibm/ceci-nest-pas-une-image:latest].
    Deleted [gcr.io/k8s-staging-cloud-provider-ibm/ceci-nest-pas-une-image@sha256:ec3ca3ee90e4dafde96c83232b30f17b5e8992ff35479d0661b8f4ff2f21bf74].
    Enabling Bucket Policy Only for gs://artifacts.k8s-staging-cloud-provider-ibm.appspot.com...
    @@ -1,3 +1,5 @@
    +- member: allUsers
    +  role: roles/storage.objectViewer
     - member: projectEditor:k8s-staging-cloud-provider-ibm
       role: roles/storage.legacyBucketOwner
     - member: projectOwner:k8s-staging-cloud-provider-ibm
    Ensuring k8s-infra-staging-cloud-provider-ibm@kubernetes.io can write to GCR for project: k8s-staging-cloud-provider-ibm
    @@ -1,5 +1,7 @@
     - member: allUsers
       role: roles/storage.objectViewer
    +- member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
    +  role: roles/storage.objectAdmin
     - member: projectEditor:k8s-staging-cloud-provider-ibm
       role: roles/storage.legacyBucketOwner
     - member: projectOwner:k8s-staging-cloud-provider-ibm
    @@ -1,5 +1,7 @@
     - member: allUsers
       role: roles/storage.objectViewer
    +- member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
    +  role: roles/storage.legacyBucketWriter
     - member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
       role: roles/storage.objectAdmin
     - member: projectEditor:k8s-staging-cloud-provider-ibm
    Ensuring GCR admins can admin GCR for project: k8s-staging-cloud-provider-ibm
    Updated IAM policy for project [k8s-staging-cloud-provider-ibm].
    @@ -1,3 +1,5 @@
    +- member: group:k8s-infra-artifact-admins@kubernetes.io
    +  role: roles/viewer
     - member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
       role: roles/viewer
     - member: serviceAccount:708000853573@cloudbuild.gserviceaccount.com
    @@ -1,5 +1,7 @@
     - member: allUsers
       role: roles/storage.objectViewer
    +- member: group:k8s-infra-artifact-admins@kubernetes.io
    +  role: roles/storage.objectAdmin
     - member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
       role: roles/storage.legacyBucketWriter
     - member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
    @@ -1,5 +1,7 @@
     - member: allUsers
       role: roles/storage.objectViewer
    +- member: group:k8s-infra-artifact-admins@kubernetes.io
    +  role: roles/storage.legacyBucketOwner
     - member: group:k8s-infra-artifact-admins@kubernetes.io
       role: roles/storage.objectAdmin
     - member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
    Ensuring GCS access logs enabled for GCR bucket in project: k8s-staging-cloud-provider-ibm
    Enabling logging on gs://artifacts.k8s-staging-cloud-provider-ibm.appspot.com/...
    @@ -1 +1 @@
    -gs://artifacts.k8s-staging-cloud-provider-ibm.appspot.com/ has no logging configuration.
    +{"logBucket": "k8s-infra-artifacts-gcslogs", "logObjectPrefix": "artifacts.k8s-staging-cloud-provider-ibm.appspot.com"}
  Ensuring staging GCS bucket: gs://k8s-staging-cloud-provider-ibm
    Ensuring gs://k8s-staging-cloud-provider-ibm exists and is world readable in project: k8s-staging-cloud-provider-ibm
    Creating gs://k8s-staging-cloud-provider-ibm/...
    Enabling Bucket Policy Only for gs://k8s-staging-cloud-provider-ibm...
    @@ -1,3 +1,5 @@
    +- member: allUsers
    +  role: roles/storage.objectViewer
     - member: projectEditor:k8s-staging-cloud-provider-ibm
       role: roles/storage.legacyBucketOwner
     - member: projectOwner:k8s-staging-cloud-provider-ibm
    Ensuring gs://k8s-staging-cloud-provider-ibm has auto-deletion of 60 days
    Setting lifecycle configuration on gs://k8s-staging-cloud-provider-ibm/...
    @@ -1 +1 @@
    -gs://k8s-staging-cloud-provider-ibm/ has no lifecycle configuration.
    +{"rule": [{"action": {"type": "Delete"}, "condition": {"age": 60}}]}
    Ensuring GCS admins can admin gs://k8s-staging-cloud-provider-ibm in project: k8s-staging-cloud-provider-ibm
    @@ -1,5 +1,7 @@
     - member: allUsers
       role: roles/storage.objectViewer
    +- member: group:k8s-infra-artifact-admins@kubernetes.io
    +  role: roles/storage.objectAdmin
     - member: projectEditor:k8s-staging-cloud-provider-ibm
       role: roles/storage.legacyBucketOwner
     - member: projectOwner:k8s-staging-cloud-provider-ibm
    @@ -1,5 +1,7 @@
     - member: allUsers
       role: roles/storage.objectViewer
    +- member: group:k8s-infra-artifact-admins@kubernetes.io
    +  role: roles/storage.legacyBucketOwner
     - member: group:k8s-infra-artifact-admins@kubernetes.io
       role: roles/storage.objectAdmin
     - member: projectEditor:k8s-staging-cloud-provider-ibm
    Ensuring k8s-infra-staging-cloud-provider-ibm@kubernetes.io can write to gs://k8s-staging-cloud-provider-ibm in project: k8s-staging-cloud-provider-ibm
    @@ -4,6 +4,8 @@
       role: roles/storage.legacyBucketOwner
     - member: group:k8s-infra-artifact-admins@kubernetes.io
       role: roles/storage.objectAdmin
    +- member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
    +  role: roles/storage.objectAdmin
     - member: projectEditor:k8s-staging-cloud-provider-ibm
       role: roles/storage.legacyBucketOwner
     - member: projectOwner:k8s-staging-cloud-provider-ibm
    @@ -4,6 +4,8 @@
       role: roles/storage.legacyBucketOwner
     - member: group:k8s-infra-artifact-admins@kubernetes.io
       role: roles/storage.objectAdmin
    +- member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
    +  role: roles/storage.legacyBucketWriter
     - member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
       role: roles/storage.objectAdmin
     - member: projectEditor:k8s-staging-cloud-provider-ibm
    Ensuring GCS access logs enabled for gs://k8s-staging-cloud-provider-ibm in project: k8s-staging-cloud-provider-ibm
    Enabling logging on gs://k8s-staging-cloud-provider-ibm/...
    @@ -1 +1 @@
    -gs://k8s-staging-cloud-provider-ibm/ has no logging configuration.
    +{"logBucket": "k8s-infra-artifacts-gcslogs", "logObjectPrefix": "k8s-staging-cloud-provider-ibm"}
  Ensuring staging GCB
    Ensuring staging bucket: gs://k8s-staging-cloud-provider-ibm-gcb
      Ensuring gs://k8s-staging-cloud-provider-ibm-gcb exists and is world readable in project: k8s-staging-cloud-provider-ibm
      Creating gs://k8s-staging-cloud-provider-ibm-gcb/...
      Enabling Bucket Policy Only for gs://k8s-staging-cloud-provider-ibm-gcb...
      @@ -1,3 +1,5 @@
      +- member: allUsers
      +  role: roles/storage.objectViewer
       - member: projectEditor:k8s-staging-cloud-provider-ibm
         role: roles/storage.legacyBucketOwner
       - member: projectOwner:k8s-staging-cloud-provider-ibm
      Ensuring gs://k8s-staging-cloud-provider-ibm-gcb has auto-deletion of 60 days
      Setting lifecycle configuration on gs://k8s-staging-cloud-provider-ibm-gcb/...
      @@ -1 +1 @@
      -gs://k8s-staging-cloud-provider-ibm-gcb/ has no lifecycle configuration.
      +{"rule": [{"action": {"type": "Delete"}, "condition": {"age": 60}}]}
      Ensuring GCS admins can admin gs://k8s-staging-cloud-provider-ibm-gcb in project: k8s-staging-cloud-provider-ibm
      @@ -1,5 +1,7 @@
       - member: allUsers
         role: roles/storage.objectViewer
      +- member: group:k8s-infra-artifact-admins@kubernetes.io
      +  role: roles/storage.objectAdmin
       - member: projectEditor:k8s-staging-cloud-provider-ibm
         role: roles/storage.legacyBucketOwner
       - member: projectOwner:k8s-staging-cloud-provider-ibm
      @@ -1,5 +1,7 @@
       - member: allUsers
         role: roles/storage.objectViewer
      +- member: group:k8s-infra-artifact-admins@kubernetes.io
      +  role: roles/storage.legacyBucketOwner
       - member: group:k8s-infra-artifact-admins@kubernetes.io
         role: roles/storage.objectAdmin
       - member: projectEditor:k8s-staging-cloud-provider-ibm
      Ensuring k8s-infra-staging-cloud-provider-ibm@kubernetes.io can write to gs://k8s-staging-cloud-provider-ibm-gcb in project: k8s-staging-cloud-provider-ibm
      @@ -4,6 +4,8 @@
         role: roles/storage.legacyBucketOwner
       - member: group:k8s-infra-artifact-admins@kubernetes.io
         role: roles/storage.objectAdmin
      +- member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
      +  role: roles/storage.objectAdmin
       - member: projectEditor:k8s-staging-cloud-provider-ibm
         role: roles/storage.legacyBucketOwner
       - member: projectOwner:k8s-staging-cloud-provider-ibm
      @@ -4,6 +4,8 @@
         role: roles/storage.legacyBucketOwner
       - member: group:k8s-infra-artifact-admins@kubernetes.io
         role: roles/storage.objectAdmin
      +- member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
      +  role: roles/storage.legacyBucketWriter
       - member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
         role: roles/storage.objectAdmin
       - member: projectEditor:k8s-staging-cloud-provider-ibm
    Ensuring k8s-infra-staging-cloud-provider-ibm@kubernetes.io can use GCB in project: k8s-staging-cloud-provider-ibm
    Updated IAM policy for project [k8s-staging-cloud-provider-ibm].
    @@ -1,5 +1,7 @@
     - member: group:k8s-infra-artifact-admins@kubernetes.io
       role: roles/viewer
    +- member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
    +  role: roles/cloudbuild.builds.editor
     - member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
       role: roles/viewer
     - member: serviceAccount:708000853573@cloudbuild.gserviceaccount.com
    Updated IAM policy for project [k8s-staging-cloud-provider-ibm].
    @@ -2,6 +2,8 @@
       role: roles/viewer
     - member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
       role: roles/cloudbuild.builds.editor
    +- member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
    +  role: roles/serviceusage.serviceUsageConsumer
     - member: group:k8s-infra-staging-cloud-provider-ibm@kubernetes.io
       role: roles/viewer
     - member: serviceAccount:708000853573@cloudbuild.gserviceaccount.com
    Ensuring gcb-builder@k8s-infra-prow-build-trusted.iam.gserviceaccount.com can use GCB in project: k8s-staging-cloud-provider-ibm
    Updated IAM policy for project [k8s-staging-cloud-provider-ibm].
    @@ -8,6 +8,8 @@
       role: roles/viewer
     - member: serviceAccount:708000853573@cloudbuild.gserviceaccount.com
       role: roles/cloudbuild.builds.builder
    +- member: serviceAccount:gcb-builder@k8s-infra-prow-build-trusted.iam.gserviceaccount.com
    +  role: roles/cloudbuild.builds.builder
     - member: serviceAccount:k8s-infra-gcr-vuln-scanning@k8s-artifacts-prod.iam.gserviceaccount.com
       role: roles/containeranalysis.occurrences.viewer
     - member: serviceAccount:service-708000853573@containerregistry.iam.gserviceaccount.com
    @@ -14,3 +14,5 @@
       role: roles/storage.legacyBucketOwner
     - member: projectViewer:k8s-staging-cloud-provider-ibm
       role: roles/storage.legacyBucketReader
    +- member: serviceAccount:gcb-builder@k8s-infra-prow-build-trusted.iam.gserviceaccount.com
    +  role: roles/storage.objectCreator
    @@ -16,3 +16,5 @@
       role: roles/storage.legacyBucketReader
     - member: serviceAccount:gcb-builder@k8s-infra-prow-build-trusted.iam.gserviceaccount.com
       role: roles/storage.objectCreator
    +- member: serviceAccount:gcb-builder@k8s-infra-prow-build-trusted.iam.gserviceaccount.com
    +  role: roles/storage.objectViewer
```

**Pull request merged**